utils/downloads.py

```diff
@@ -53,3 +54,30 @@ def download_sounds(licenses_url, pack):
     response = HttpResponse(filelist, content_type="text/plain")
     response['X-Archive-Files'] = 'zip'
     return response
+
+
+def should_suggest_donation(user, times_displayed):
+    """
+    This method indicates when we should display the donation popup to the
+    user. This will be based on 3 settings indicating how many days
+    after a donation we show the popup again, after how many downloads we
+    display the popup and for how long. The popup will be shown a number of
+    times per day.
+    """
+
+    if times_displayed and times_displayed > settings.DONATION_MODAL_DISPLAY_TIMES_DAY:
+        return False
+
+    last_donation = user.donation_set.order_by('created').last()
+    period = datetime.datetime.now()\
+            - datetime.timedelta(days=settings.DONATION_MODAL_DOWNLOAD_DAYS)
+    num_downloads_in_period = Download.objects.filter(user=user,
```

Comment:
So, the number of downloads are counted in a cookie, but on top of that we're also doing a downloads-count via the Database on each get() triggered by each download? It seems to me this either warrants doing one OR the other, but not both.